packages/xy-chart/src/selection/Brush.jsx

```diff
@@ -95,24 +116,64 @@ class Brush extends React.Component {
   convertRangeToDomain(brush) {
     const { xScale, yScale } = this.props;
     const { x0, x1, y0, y1 } = brush.extent;
-    const invertedX0 = xScale.invert(x0 + (x0 < x1 ? -SAFE_PIXEL : SAFE_PIXEL));
-    const invertedX1 = xScale.invert(x1 + (x1 < x0 ? -SAFE_PIXEL : SAFE_PIXEL));
-    const invertedY0 = yScale.invert(y0 + (y0 < y1 ? -SAFE_PIXEL : SAFE_PIXEL));
-    const invertedY1 = yScale.invert(y1 + (y1 < y0 ? -SAFE_PIXEL : SAFE_PIXEL));
+
+    let xDomain;
+    const invertedX0 = scaleInvert(xScale, x0 + (x0 < x1 ? -SAFE_PIXEL : SAFE_PIXEL));
+    const invertedX1 = scaleInvert(xScale, x1 + (x1 < x0 ? -SAFE_PIXEL : SAFE_PIXEL));
+    const startX = Math.min(invertedX0, invertedX1);
+    const endX = Math.max(invertedX0, invertedX1);
+    if (xScale.invert) {
+      xDomain = {
+        x0: startX,
+        x1: endX,
+      };
+    } else {
+      const xValues = [];
+      for (let i = startX; i <= endX; i += 1) {
+        xValues.push(xScale.domain()[i]);
```

Comment:
could call `.domain` once and assign to a variable here instead of calling multiple times.